### PR TITLE
fix: Distinguish between hits and measurements in `ParticleSelector` in Examples

### PR DESCRIPTION
**Pull request by andiwand**
This clears out an oversight in a previous PR https://github.com/acts-project/acts/pull/3742 where I replaced numbers of measurements with number of hits. This number can be different and should not be confused.

I add an optional input for the particle to measurement map which is then queried if present and used for the number of measurement cuts.

blocked by:
- https://github.com/acts-project/acts/pull/3944

## Release Notes

- **New Features**
  - Enhanced particle selection with new configuration options for minimum and maximum hit counts.
  - Updated particle selection criteria to focus on hit counts instead of measurement counts.

- **Bug Fixes**
  - Improved validation checks for input measurements to ensure proper initialization.

- **Documentation**
  - Updated comments and documentation to reflect changes in parameter naming from "measurements" to "hits."

- **Tests**
  - Adjusted test cases to align with the new parameter naming conventions.

- **Chores**
  - Minor code formatting and organization improvements throughout various scripts.

### CODE REVIEW

File: Examples/Algorithms/TruthTracking/ActsExamples/TruthTracking/ParticleSelector.cpp [hunk 1]
```diff
@@ -28,8 +30,17 @@ ActsExamples::ParticleSelector::ParticleSelector(const Config& config,
   }
 
   m_inputParticles.initialize(m_cfg.inputParticles);
+  m_inputParticleMeasurementsMap.maybeInitialize(
+      m_cfg.inputParticleMeasurementsMap);
   m_outputParticles.initialize(m_cfg.outputParticles);
 
+  if (m_inputParticleMeasurementsMap.isInitialized() &&
```
Comment (benjaminhuth) [inline Examples/Algorithms/TruthTracking/ActsExamples/TruthTracking/ParticleSelector.cpp:37]: Is here a negation missing?

**Review by benjaminhuth — COMMENTED** [review]
Looks good, I have one comment though.

File: Examples/Algorithms/TruthTracking/ActsExamples/TruthTracking/ParticleSelector.cpp (hunk 1 above)
Comment (andiwand) [reply Examples/Algorithms/TruthTracking/ActsExamples/TruthTracking/ParticleSelector.cpp:37]: nice catch!

Comment (andiwand) [reply Examples/Algorithms/TruthTracking/ActsExamples/TruthTracking/ParticleSelector.cpp:37]:
```suggestion
  if (!m_inputParticleMeasurementsMap.isInitialized() &&
```

**Review by benjaminhuth — APPROVED** [review]
Looks good!

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results [03315f87b1c6a1d4ff17341e160f484753af81e1]
### :white_check_mark: All tests successful

|status|job|report|
|------|---|------|
|:green_circle: | [report_pull_request](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47739820)||
|:yellow_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47739819)||
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47739818)||
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47739817)||
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47739816)||
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47739815)||
|:green_circle: | [test_ActsDumpGeometryIdentifiers](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47739814)||
|:green_circle: | [test_ActsCheckObjectCountsWorkflowHgtd](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47739813)||
|:green_circle: | [test_ActsCheckObjectCountsCachedWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47739812)||
|:green_circle: | [test_ActsCheckObjectCountsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47739811)||
|:green_circle: | [test_ActsEFTrackFit](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47739810)||
|:green_circle: | [test_ActsPersistifySeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47739809)||
|:green_circle: | [test_ActsBenchmarkWithSpot](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47739808)||
|:green_circle: | [test_ActsAnalogueClustering](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47739807)||
|:green_circle: | [test_ActsWorkflowHeavyIons](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47739806)||
|:green_circle: | [test_ActsWorkflowFastTracking](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47739805)||
|:green_circle: | [test_ActsWorkflowCached](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47739804)||
|:green_circle: | [test_ActsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47739803)||
|:green_circle: | [test_ActsValidateAmbiguityResolution](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47739802)||
|:green_circle: | [test_ActsValidateResolvedTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47739801)||
|:green_circle: | [test_ActsValidateTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47739800)||
|:green_circle: | [test_ActsValidateActsCoreSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47739799)||
|:green_circle: | [test_ActsValidateActsSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47739798)||
|:green_circle: | [test_ActsValidateSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47739797)||
|:green_circle: | [test_ActsValidateOrthogonalSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47739796)||
|:green_circle: | [test_ActsValidateClusters](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47739795)||
|:green_circle: | [test_ActsPersistifyEDM](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47739794)||
|:green_circle: | [test_ActsGx2fRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47739793)||
|:green_circle: | [test_ActsGSFRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47739792)||
|:green_circle: | [test_ActsKfRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47739791)||
|:green_circle: | [test_ActsExtrapolationAlgTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47739790)||
|:green_circle: | [test_ActsITkTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47739789)||

**Pull request merged**